class-work/02-deploy-python.qmd

```diff
@@ -0,0 +1,129 @@
+---
+title: Deploy your model
+subtitle: Deploy and maintain models with vetiver (Python)
+format: html
+---
+
+## Create a new vetiver model 🌳🌴🌲🌲🌴🌳🌴🌳🌲
+
+Fit a random forest model
+
+```{python}
+import pandas as pd
+import numpy as np
+from sklearn import preprocessing, ensemble, pipeline, compose, model_selection
+
+inspections = pd.read_parquet('../data/inspections.parquet')
+inspections['inspection_date'] = pd.to_datetime(inspections['inspection_date'])
+inspections['month'] = inspections['inspection_date'].dt.month
+inspections['year'] = inspections['inspection_date'].dt.year
+
+categorical_features = ['facility_type', 'risk', 'month']
+X, y = inspections.drop(columns=['aka_name', 'results','inspection_date']), inspections['results']
+X_train, X_test, y_train, y_test = model_selection.train_test_split(
+    X, y,
+    stratify = y,
+    test_size = 0.2
+)
+
+oe = compose.make_column_transformer(
+    (preprocessing.OrdinalEncoder(
+            handle_unknown="use_encoded_value", unknown_value=-1),
+        categorical_features,),
+    remainder="passthrough",
+).fit(X_train)
+rf = ensemble.RandomForestClassifier().fit(oe.transform(X_train), y_train)
+inspection_fit = pipeline.Pipeline([("ordinal_encoder", oe), ("random_forest", rf)])
+```
+
+Store your model on Posit Connect:
+
+```{python}
+#| eval: false
+from vetiver import VetiverModel, vetiver_pin_write
+from pins import board_connect
+import os
+from dotenv import load_dotenv
+load_dotenv()
+api_key = os.getenv("CO_API_KEY")
+rsc_url = os.getenv("CO_RSC_URL")
+
+v = VetiverModel(__)
+board = ___
+vetiver_pin_write(board, v)
+```
+
+## Model Card ❤️
+
+Open the Model Card template in RStudio by choosing "File" ➡️ "New File" ➡️ "R Markdown" ➡️ "From Template" ➡️ "Vetiver Model Card".  
```

Comment:
@isabelizimm can you update this for the best way for Python folks to get the Model Card template? Is it just `vetiver.model_card(path = ".")`?

Reply:
done!